generator/src/test/java/com/linkedin/pegasus/generator/TestPegasusDataTemplateGenerator.java

```diff
@@ -205,4 +208,123 @@ private void testRunGenerator(String pegasusFilename, Map<String, String> expect
             file -> file.getName().replace(".java", ""),
             Function.identity()));
   }
+
+  @DataProvider(name = "with_typeref_field_in_record_pdsc")
+  private Object[][] createDataTemplateCases()
+  {
+    return new Object[][]
+        {
+            {"IsoDuration.pdsc", "PremiumService.pdsc", "IsoDuration.java", "PremiumService.java", "PremiumServiceState.java", "AccountBalanceType.java"},
+            {"PremiumService.pdsc", "IsoDuration.pdsc", "IsoDuration.java", "PremiumService.java", "PremiumServiceState.java", "AccountBalanceType.java"},
+        };
+  }
+
+  @Test(dataProvider = "with_typeref_field_in_record_pdsc")
+  public void testDataTemplateGenerationOrderWithResolver(String[] testArgs)
```

Comment:
Oh, sorry, I just realized what you are doing with all these tests. 

Since you only really have one test with a nuance on the inputs, the better practice would be to have 1 DataProvider with your different tests as it's inputs. 

i.e. 

data resolver
{
  // typered fields with pdsc files
  { ...} 
  {...}
  // typeref field with pdl files
 { ...}
 { ... } 
 // inline field with pdsc files
 { ...} 
 {...}
 //inline field with pdl files
 { ...}
 { ...}
}

then your test would be just the one test you had.

Reply:
Yes, thanks, addressed into a single data provider "test_schema_permutation_deterministic"

Reply:
I see that you changed to a single data provider but you still have only 1 input arrray into your test, then within your test, you split it up by dividing by 2. 

You should have two separate arrays, one for each variable. It makes it a lot easier to understand the logic within your code/test.